tests/mockdata.py

```diff
@@ -27,36 +25,34 @@ def generate_fake_rastercube(
         spatial_extent.south, spatial_extent.north
     )
 
-    step_size = max(len_x, len_y) / n_spatial_coords
-
-    x = np.arange(
+    x_coords = np.arange(
         min(spatial_extent.west, spatial_extent.east),
         max(spatial_extent.west, spatial_extent.east),
-        step=step_size,
+        step=len_x / data.shape[0],
     )
-    y = np.arange(
+    y_coords = np.arange(
         min(spatial_extent.south, spatial_extent.north),
         max(spatial_extent.south, spatial_extent.north),
-        step=step_size,
+        step=len_y / data.shape[1],
     )
-    t = pd.date_range(
+    t_coords = pd.date_range(
         start=np.datetime64(temporal_extent.__root__[0].__root__),
         end=np.datetime64(temporal_extent.__root__[1].__root__),
-        periods=n_timesteps,
+        periods=data.shape[2],
     ).values
 
-    coords = {"x": x, "y": y, "t": t, "bands": bands}
-
-    # This is to enable simulating fake data from different collections.
-    # The [:9] part is necessary because Dask.random.seed can only accept 32-bit values
-    da.random.seed(int(str(abs(hash(seed)))[:9]))
-    _data = da.random.random(tuple([len(v) for _, v in coords.items()]))
+    coords = {"x": x_coords, "y": y_coords, "t": t_coords, "bands": bands}
 
-    data = xr.DataArray(
-        data=_data,
+    raster_cube = xr.DataArray(
+        data=data,
         coords=coords,
         attrs={"crs": spatial_extent.crs},
```

Comment:
keep meaning to mention, but we probably shoudn't set data crs values to actualy CRS objects, either the wkt string or just the epsg code is enough and usable, the CRS object will cause an exception when trying to write the data to a netcdf or a gtiff, and imo it's a bit annoying to reset the values to string when wanting to write. Wdyut? If we agree might be worth actually changing this upstream in the parser too so it's clearer throughout.

Reply:
Ahh, didn't realise that was a problem! That's a fair point, we can just use the CRS object during validation, but then store the wkt string going forward!